benchmark/utils/generator.hpp

```diff
@@ -55,23 +55,56 @@ struct DefaultSystemGenerator {
     static gko::matrix_data<ValueType, IndexType> generate_matrix_data(
         const json& config)
     {
+        gko::matrix_data<ValueType, IndexType> data;
         if (config.contains("filename")) {
             std::ifstream in(config["filename"].get<std::string>());
-            return gko::read_generic_raw<ValueType, IndexType>(in);
+            data = gko::read_generic_raw<ValueType, IndexType>(in);
         } else if (config.contains("stencil")) {
-            return generate_stencil<ValueType, IndexType>(
+            data = generate_stencil<ValueType, IndexType>(
                 config["stencil"].get<std::string>(),
                 config["size"].get<gko::int64>());
         } else {
             throw std::runtime_error(
                 "No known way to generate matrix data found.");
         }
+        if (config.contains("reorder")) {
+            using Csr = gko::matrix::Csr<ValueType, IndexType>;
+            auto ref = gko::ReferenceExecutor::create();
+            auto mtx = gko::share(Csr::create(ref));
+            mtx->read(data);
+            auto reorder_type = config["reorder"].get<std::string>();
+            if (reorder_type == "amd") {
+                auto perm = gko::experimental::reorder::Amd<IndexType>::build()
+                                .on(ref)
+                                ->generate(mtx);
```

Comment:
I think it makes sense to measure the time it takes to reorder as well. It does not have to be here, but could be added as a separate part within the detailed benchmark.

Reply:
This is already possible through `sparse_blas`, I would like to avoid supporting this in every benchmark executable, since it mixes the setup phase with the benchmarking phase. Does requiring the users to execute two benchmarks sound acceptable?

Reply:
I think two benchmarks should be okay.